src/HttpHeaderTools.cc

```diff
@@ -233,6 +234,19 @@ httpHeaderParseQuotedString(const char *start, const int len, String *val)
     return 1;
 }
 
+/**
+ * Temporary bridge for the String-based variant to support conversion
+ * from SquidString to SBuf
+ * TODO: make this the primary implementation
+ */
+int httpHeaderParseQuotedString(const char *start, const int len, SBuf &val)
```

Comment:
Most likely, the long-term "primary" function should have a different API. For example, I would expect it return an SBuf (and an exception thrown on errors, as usual). The closer you can get to that correct long-term interface in this PR, when you are adding the very first callers for that function, the better! Otherwise, we would be creating new technical debt (i.e. wrong HTTP quoted string parsing API) while trying to get rid of the old technical debt (i.e. String).

Reply:
Done in the SBuf-based variant